File: 49-group-anagrams/group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> hmap = new HashMap<>();

        for (String str : strs) {
            char[] charArray = str.toCharArray();
            Arrays.sort(charArray);

            String key = new String(charArray);

            if (hmap.containsKey(key)) {
                hmap.get(key).add(str);
            } else {
                List<String> list = new ArrayList<>();
                list.add(str);
                hmap.put(key, list);
            }
        }

        return new ArrayList<>(hmap.values());
    }
}

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const  map = new Map();
    for(let i=0;i<strs.length;i++){
        const s =strs[i].split("").sort().join("");
        map[s] = [];
    }
    for(let i=0;i<strs.length;i++){
          const s =strs[i].split("").sort().join("");
           map[s].push(strs[i])
    }
    
    return Object.values(map);
    
};